backend/api/v1/routers/admin_structures.py

```diff
@@ -35,61 +34,80 @@ async def create_admin_structure(
-            username=username,
-        )
-
-        if not account:
-            logging.error(f"Insert account failed")
-            raise HTTPException(status_code=500, detail="insert account failed")
-
-        ass_id = await insert_admin_structure_structure(
-            connection=db,
-            admin_structure_id=admin_structure.id,
-            structure_id=data.structure_id,
-        )
-
-        if not ass_id:
-            logging.error(f"Insert admin_structure_structure failed")
-            raise HTTPException(
-                status_code=500,
-                detail="insert admin_structure_structure failed",
+        if not admin_structure:
+            try:
+                account_admin_tuple: Tuple[
+                    AccountDB, AdminStructure
+                ] = await create_admin_structure_with_account(db, data)
+                account, admin_structure = account_admin_tuple
+
+                background_tasks.add_task(
+                    send_invitation_email,
+                    email=admin_structure.email,
+                    firstname=admin_structure.firstname,
+                    lastname=admin_structure.lastname,
+                    access_key=account.access_key,
+                )
+
+            except InsertFailError as error:
+                logging.error(error)
+                raise HTTPException(
+                    status_code=500,
+                    detail=error,
+                ) from error
+
+            except Exception as error:
+                logging.error(error)
+                raise HTTPException(
+                    status_code=500, detail="fail to create admin structure structure"
+                ) from error
+        try:
+
+            ass_id: UUID | None = await insert_admin_structure_structure(
+                connection=db,
+                admin_structure_id=admin_structure.id,
+                structure_id=data.structure_id,
             )
 
-        background_tasks.add_task(
-            send_invitation_email,
-            email=admin_structure.email,
-            firstname=admin_structure.firstname,
-            lastname=admin_structure.lastname,
-            access_key=account.access_key,
-        )
-
-        return admin_structure
+            if not ass_id:
+                logging.error(f"Insert admin_structure_structure failed")
+                raise HTTPException(
+                    status_code=500,
+                    detail="insert admin_structure_structure failed",
+                )
+            return admin_structure
+
+        except UniqueViolationError as error:
+            logging.error(
+                "uniqueness violation fail for insert admin_structure_structure: {}".format(
+                    error
+                )
+            )
+            raise HTTPException(
+                status_code=422,
+                detail="duplicate key value violates unique constraint admin_structure_structure_admin_structure_id_structure_id_key",
```

Comment:
Je pinaille mais je pense que l'erreur Http ne devrait pas mentionner la violation de contrainte de la base de données.
Je propose :
```suggestion
                detail="impossible to associate given admin with given structure: relationship already exists",
```

Reply:
💯